File: Shortest_Dist/dist.go
package shortestdist

func shortestToChar(s string, c byte) []int {
	result := []int{}

	for k := 0; k < len(s); k++ {
		if s[k] == c {
			result = append(result, 0)
		} else {
			i, j := k-1, k+1

			for i >= 0 && s[i] != c {
				i--
			}

			for j < len(s) && s[j] != c {
				j++
			}

			minDist := getMin(k-i, j-k)
			if i == -1 {
				minDist = j - k
			}

			if j == len(s) {
				minDist = k - i
			}
			result = append(result, minDist)
		}
	}

	return result
}

func getMin(a, b int) int {
	if a > b {
		return b
	}

	return a
}

File: Shortest_Dist/dist.js
/**
 * @param {string} s
 * @param {character} c
 * @returns {number[]}
 */
var shortestToChar = function (s, c) {
	const result = [];

	for (let k = 0; k < s.length; k++) {
		if (s[k] === c) {
			result.push(0);
		} else {
			let [i, j] = [k - 1, k + 1];

			while (i >= 0 && s[i] !== c) {
				i--;
			}

			while (j < s.length && s[j] !== c) {
				j++;
			}

			let minDist = Math.min(k - i, j - k);

			if (i == -1) {
				minDist = j - k;
			}

			if (j == s.length) {
				minDist = k - i;
			}

			result.push(minDist);
		}
	}
	return result;
};
